src/program.fs

```diff
@@ -87,31 +87,35 @@ module Program =
     /// program: program created with 'mkSimple' or 'mkProgram'.
     let runWith (arg: 'arg) (program: Program<'arg, 'model, 'msg, 'view>) =
         let (model,cmd) = program.init arg
-        let inbox = MailboxProcessor.Start(fun (mb:MailboxProcessor<'msg>) ->
-            let rec loop (state:'model) =
-                async {
-                    let! msg = mb.Receive()
-                    let newState =
-                        try
-                            let (model',cmd') = program.update msg state
-                            program.setState model' mb.Post
-                            cmd' |> Cmd.exec mb.Post
-                            model'
-                        with ex ->
-                            program.onError ("Unable to process a message:", ex)
-                            state
-                    return! loop newState
-                }
-            loop model
-        )
-        program.setState model inbox.Post
+        let mutable rb = RingBuffer 10
+        let mutable reentered = false
```

Comment:
flag indicating that dispatch is entered recursively via a cmd that was processed as part of earlier entry in the stack.